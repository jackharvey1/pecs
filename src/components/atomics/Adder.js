import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { MdAdd } from 'react-icons/md';

const DottedSquare = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;

    height: 50px;
    width: 50px;

    font-size: 40px;

    border-style: dashed;
    border-color: lightgray;

    &:first-child {
        margin-left: auto;
    }

    &:last-child {
        margin-right: auto;
    }
`;

const Adder = ({ showGallery, index }) => (
    <DottedSquare onClick={() => showGallery(index)}>
        <MdAdd />
    </DottedSquare>
);

Adder.propTypes = {
    showGallery: PropTypes.func.isRequired,
    index: PropTypes.number.isRequired,
};

export default Adder;
