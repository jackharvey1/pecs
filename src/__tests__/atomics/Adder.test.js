import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Adder from '../../components/atomics/Adder';

describe('Adder component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(
            <Adder
                index={0}
                showGallery={() => null}
            />
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('calls showGallery with the correct inded when clicked', () => {
        const mockShowGallery = jest.fn();
        const wrapper = shallow(
            <Adder
                index={0}
                showGallery={mockShowGallery}
            />
        );

        wrapper.simulate('click');

        expect(mockShowGallery).toHaveBeenCalledWith(0);
    });
});
