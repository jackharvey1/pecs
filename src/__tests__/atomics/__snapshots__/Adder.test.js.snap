// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adder component renders as intended 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 50px;
  width: 50px;
  font-size: 40px;
  border-style: dashed;
  border-color: lightgray;
}

.c0:first-child {
  margin-left: auto;
}

.c0:last-child {
  margin-right: auto;
}

<div
  className="c0"
  onClick={[Function]}
>
  <svg
    className={undefined}
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</div>
`;
